src/App.tsx

```diff
@@ -1,39 +1,64 @@
 import React from 'react';
 import './App.scss';
+import { Clock } from './component/Clock';
 
-function getRandomName(): string {
-  const value = Date.now().toString().slice(-4);
+// let clockName = 'Clock-0';
+type State = {
+  clockName: string,
+  hasClock: boolean,
+};
 
-  return `Clock-${value}`;
-}
+export class App extends React.PureComponent {
+  state: State = {
+    clockName: 'Clock-0',
+    hasClock: true,
+  };
+
+  timerId = 0;
+
+  componentDidMount(): void {
+    function getRandomName() {
+      const value = Date.now().toString().slice(-4);
+
+      return `Clock-${value}`;
+    }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+    this.timerId = window.setInterval(() => {
+      this.setState({ clockName: getRandomName() });
+    }, 3300);
```

Comment:
```suggestion
    this.timerId = window.setInterval(() => {
      this.setState({ clockName: this.getRandomName() });
    }, 3300);
```
And then use it like this.